### PR TITLE
refactoring - better error handling and pylint enabling

### PR DESCRIPTION
**Pull request by stopal-r7**
This refactors apiutils for better error handling and enables integrated pylint.

### CODE REVIEW

**Comment by StephenHynes7** [comment]
Still failing on travis : (

**Comment by stopal-r7** [comment]
Yep trying to figure out what it is. Enabling pylint has its consequences :)

**Comment by stopal-r7** [comment]
Added pylint to requirement file, should be OK now.

File: lecli/apiutils.py [hunk 1]
```diff
@@ -35,103 +60,116 @@ def init_config():
...
-    ro_apikey = None
+    section = 'Auth'
+    config_key = 'ro_api_key'
     try:
-        ro_apikey = CONFIG.get('Auth', 'ro_api_key')
-        if len(ro_apikey) != 36:
-            print 'Error: Read-only API Key not of correct length'
+        ro_api_key = CONFIG.get(section, config_key)
+        if len(ro_api_key) != 36:
+            print_config_error_and_exit(section, 'Read-only API key(%s)' % config_key, ro_api_key)
```
Comment (pclune-r7) [inline lecli/apiutils.py]: should there be a second %s as you have two variables after the % ?

File: lecli/apiutils.py [hunk 2]
```diff
@@ -35,103 +60,116 @@ def init_config():
...
-    rw_apikey = None
+    section = 'Auth'
+    config_key = 'rw_api_key'
     try:
-        rw_apikey = CONFIG.get('Auth', 'rw_api_key')
-        if len(rw_apikey) != 36:
-            print 'Error: Read/Write API Key not of correct length'
+        rw_api_key = CONFIG.get(section, config_key)
+        if len(rw_api_key) != 36:
+            print_config_error_and_exit(section, 'Read/Write API key(%s)' % config_key, rw_api_key)
```
Comment (pclune-r7) [inline lecli/apiutils.py]: same as above here

Comment (pclune-r7) [inline lecli/apiutils.py]: forget it - my mistake
no change needed

Comment (stopal-r7) [inline lecli/apiutils.py]: Mutliple usage of Auth section caught my attention, so declared it as global. This is clearer I think.

**Comment by pclune-r7** [comment]
+1

**Pull request merged**